homeassistant/components/shelly/button.py

```diff
@@ -270,3 +309,32 @@ async def _press_method(self) -> None:
             assert method is not None
 
         await method(self._id)
+
+
+class ShellyVirtualButton(ShellyBaseButton):
+    """Defines a Shelly virtual component button."""
+
+    def __init__(
+        self,
+        coordinator: ShellyRpcCoordinator,
+        description: ShellyButtonDescription,
+        _id: int,
+    ) -> None:
+        """Initialize Shelly virtual component button."""
+        super().__init__(coordinator, description)
+
+        self._attr_unique_id = f"{coordinator.mac}-{description.key}:{_id}-button"
```

Comment:
I don't think we need the `-button` suffix. `123456789ABC-button:200` is enough unique.

Reply:
I was not sure if I should add it or not, I will remove it and update the PR